subworkflows/local/binning_refinement.nf

```diff
@@ -41,30 +42,38 @@ workflow BINNING_REFINEMENT {
                                         .branch {
                                             metabat2: it[0]['binner'] == 'MetaBAT2'
                                             maxbin2:  it[0]['binner'] == 'MaxBin2'
+                                            concoct: it[0]['binner'] == 'CONCOCT'
                                         }
 
     // Generate DASTool auxilary files
     DASTOOL_FASTATOCONTIG2BIN_METABAT2 ( ch_bins_for_fastatocontig2bin.metabat2, "fa")
     // MaxBin2 bin extension was changed to 'fa' as well in RENAME_PREDASTOOL
     DASTOOL_FASTATOCONTIG2BIN_MAXBIN2 ( ch_bins_for_fastatocontig2bin.maxbin2, "fa")
+    DASTOOL_FASTATOCONTIG2BIN_CONCOCT ( ch_bins_for_fastatocontig2bin.concoct, "fa")
 
     // Run DASTOOL
     ch_fastatocontig2bin_for_dastool = Channel.empty()
     ch_fastatocontig2bin_for_dastool = ch_fastatocontig2bin_for_dastool
                                     .mix(DASTOOL_FASTATOCONTIG2BIN_METABAT2.out.fastatocontig2bin)
                                     .mix(DASTOOL_FASTATOCONTIG2BIN_MAXBIN2.out.fastatocontig2bin)
+                                    .mix(DASTOOL_FASTATOCONTIG2BIN_CONCOCT.out.fastatocontig2bin)
                                     .map {
                                         meta, fastatocontig2bin ->
                                             def meta_new = meta.clone()
                                             meta_new.remove('binner')
                                             [ meta_new, fastatocontig2bin ]
                                     }
                                     .groupTuple(by: 0)
+                                    .dump(tag: "fastatocontigbin_for_dastool")
+                                    // TODO: contigs names might be truncated? https://github.com/cmks/DAS_Tool/issues/78#issuecomment-1086543966
+                                    // Error: Contigs of contig2bin files not found in assembly: 36
+                                    // /home/jfellows/Documents/git/nf-core/mag/testing/work/77/89fbe4f1d985f06ebfac3816b87560
```

Comment:
Is that still a problem?

Reply:
It wasn't an issue with our pipeline but a bug in DAS_Tool and in this case only effects CONCOCT:

https://github.com/cmks/DAS_Tool/issues/90

I'm waiting to see if there will be a quick patch release for it in the next day or so (I wanted the rest of this PR reviewed in the meantime), if not I'll go back and fix the header names coming out of CONCOCT so they don't have spaces in them.

I just noticed I forgot to put this to draft... will do that now

Reply:
While there is some activity on DAS_Tool I don't know how fast it will be. Instead I will 'fix' the MEGAHIT contig names instead

Reply:
I've added a 'fix' for the MEGAHIT contig names (which ended up being more complicated due to PROKKA contig length name restrictions...)